src/processors/file-handler.js

```diff
@@ -112,8 +113,9 @@ export async function* readBatch(asyncIterator, fileName) {
       if (batch.jsonpath && batch.jsonpath.length > 1) {
         const streamingPath = new _JSONPath(batch.jsonpath);
         streamingPath.setFieldAtPath(result, batches);
-      } else {
-        // The streamed object is a ROW JSON-batch
+      } else if (batch.jsonpath && batch.jsonpath.length === 1) {
+        // The streamed object is a ROW JSON-batch (jsonpath = '$')
+        // row objects
         result = batches;
```

Comment:
It is a bit confusing to call this variable `batches`, because in my mind we are iterating over the batches and accumulating a data array/table.

I would just call this `data` (or if more elaborate perhaps `streamedData` or `accumulatedData` or `table` etc).